docs/source/conf.py

```diff
@@ -41,6 +49,20 @@
             check=True,
         )
 
+# Sphinx is very finicky, and somewhat buggy, so we have several different
+# methods to help it resolve links.
```

Comment:
Might want to mention the `typevar` module also, which redirects `typing_extensions` → `typing`, and all typevars to `typing.TypeVar` etc.